### ISSUE TITLE
Database is not initialized for in-memory db

### ISSUE THREAD

**Issue by perama-v** [pipermerriam/glados#30]
### Description

In `glados-monitor`, when an in-memory database is created using the `"sqlite::memory:"` argument, as in:

```command
$ cargo run -p glados-monitor -- --database-url sqlite::memory: follow-head --provider-url http://127.0.0.1:8545
```
The following error is encountered:
```sh
Query(SqlxError(Database(SqliteError { code: 1, message: "no such table: content_key" })))
```
path
glados/entity/src/contentkey.rs:46:10
```
This indicates that upon lookup of the first key in the table, the action fails as the result of a non-existent table. 
```rs
// glados/entity/src/contentkey.rs
get_or_create(content_key_raw: &impl ContentKey, conn: &DatabaseConnection) -> Model {
    // First try to lookup an existing entry.
    let content_key = Entity::find()
        .filter(Column::ContentKey.eq(content_key_raw.encode()))
        .one(conn)
        .await
        .unwrap(); <--- panic
    // snip
}
```
## Resolution
Some options is see:
1. Require that the CLI flags prevent in-memory-db + no-migration combination
2. Run db migration if in-memory-db is selected (this initialises the db)

**Comment by mrferris** [comment]
Is this issue specific to in memory DB? I'd think this same thing happens for a sqlite DB.

I think that both options 1 & 2 would remove the control that the `migrate` flag gives us.

I propose options:

3. Detect whether there's any DB, if so always run `Migrator::up`. If a DB exists, check the `migrate` flag.
4. Just pass `--migrate` in every time, unless you specifically don't want to migrate.
5. Replace `--migrate` with `--no-migrate` and make `Migrator::up` the default behavior.

**Comment by perama-v** [comment]
Good point, I've updated #31 to do `3` instead. The use of the migration flag for a new or in-memory database (`4`) feels counter-intuitive and agree on `5`.

**Comment by perama-v** [comment]
On discussion in the PR, favouring `4` at this point, with the addition that in-memory db always triggers `Migrator::up()`

**Comment by perama-v** [comment]
Closed by #31
